apps/system/js/app_chrome.js

```diff
@@ -515,14 +515,19 @@
     }
 
     var self = this;
-    var previousColor;
+    var finishedFade = false;
+    var endBackgroundFade = function() {
+      finishedFade = true;
+      self.element.removeEventListener('transitionend', endBackgroundFade);
+    };
+    this.element.addEventListener('transitionend', endBackgroundFade);
 
     window.requestAnimationFrame(function updateAppColor() {
```

Comment:
I'd also question the importance of the timing in that the requestAnimationFrame isn't going to be perfectly in sync with the async animation, so there's not much point in trying to be so accurate about it (and in fact, working it out beforehand and using setTimeout may produce a better result...)